### PR TITLE
Fix Phpdoc

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

BC

Should solve the psalm issue in https://github.com/sonata-project/SonataDoctrineORMAdminBundle/pull/1573/files

## Changelog

```markdown
### Fixed
- Filter interface phpdoc
```

### CODE REVIEW

File: src/Filter/FilterInterface.php [hunk 1]
```diff
@@ -90,14 +90,14 @@ public function getParentAssociationMappings(): array;
     /**
      * @throws \RuntimeException if the `field_mapping` option is not set
      *
-     * @return array<string, string> field mapping
+     * @return array<string, mixed> field mapping
```
Comment (core23) [inline src/Filter/FilterInterface.php:93]: Can we add an array shape signature here?

Comment (VincentLanglet) [reply src/Filter/FilterInterface.php:93]: I would say no because it's persistence bundle specific.

Comment (jordisala1991) [reply src/Filter/FilterInterface.php:93]: Even if it is persistence bundle specific, this interface defines the contract. What options are normally returned here?

Comment (VincentLanglet) [reply src/Filter/FilterInterface.php:93]: I would recommend to discuss this in an issue like https://github.com/sonata-project/SonataAdminBundle/issues/6274 and not block this PR which is just a phpdoc fix.

The option returns can be anything, based on what the persistence bundle use. DoctrineORM and DoctrineMongo doesn't use the same key or values. For instance targetDocument is used in mongo but not in orm. And I could write my own persistence bundle with some others keys.

So currently any keys or values are allowed and IMHO should be allowed for now until something better is proposed like an Object

**Review by jordisala1991 — APPROVED** [review]

**Review by core23 — APPROVED** [review]

**Pull request merged**